sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -490,9 +495,15 @@ class Analyzer(
 
       // If the aggregate function argument contains Stars, expand it.
       case a: Aggregate if containsStar(a.aggregateExpressions) =>
-        val expanded = expandStarExpressions(a.aggregateExpressions, a.child)
+        if (conf.groupByOrdinal && a.groupingExpressions.exists(IntegerIndex.unapply(_).nonEmpty)) {
```

Comment:
Actually, the major reason why I set default to `true` is `ordinal number` in Group/Order by has no effect. Here, I assume users are normally doing something reasonable. When users put `ordinal number` in Order/Group by, normally, they expect they are meaningful. Thus, I suspect they want to use `ordinal positions`.

The reason why Hive is set the default to false. I think that is to keep the migration easier. Make the versions consistent. Spark 2.0 makes a lot of external changes. Thus, I think it is OK to change the behaviors of ordinal number resolution.